File: frontend/src/microservices/train/objects/BookTrainItemPlannerRequest.ts
// 本文件定义 train 模块的 `BookTrainItemPlannerRequest`，作为与 planner 同名的请求参数并提供 JSON 编解码。

export type BookTrainItemPlannerRequest = {
  userId: string
  trainId: string
  travelerIds: string[]
  fromStationCode: string
  toStationCode: string
  seatClass: string
  seatPreference?: string | null
}

export const bookTrainItemPlannerRequestFromJson = (json: string): BookTrainItemPlannerRequest =>
  JSON.parse(json) as BookTrainItemPlannerRequest

export const bookTrainItemPlannerRequestToJson = (value: BookTrainItemPlannerRequest): string =>
  JSON.stringify(value)

File: backend/src/main/scala/microservices/train/objects/BookTrainItemPlannerRequest.scala
package com.typesafe.travel.train.domain

import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}

final case class BookTrainItemPlannerRequest(
    userId: String,
    orderId: String,
    trainId: String,
    travelerIds: List[String],
    fromStationCode: String,
    toStationCode: String,
    seatClass: String,
    seatPreference: Option[String]
)

object BookTrainItemPlannerRequest:
  given sourceEncoder: Encoder[BookTrainItemPlannerRequest] = deriveEncoder
  given sourceDecoder: Decoder[BookTrainItemPlannerRequest] = deriveDecoder
